hudi-common/src/main/java/org/apache/hudi/common/model/HoodieCommitMetadata.java

```diff
@@ -236,6 +241,43 @@ public static <T> T fromJsonString(String jsonStr, Class<T> clazz) throws Except
     return JsonUtils.getObjectMapper().readValue(jsonStr, clazz);
   }
 
+  /**
+   * parse the bytes of deltacommit, and get the base file and the log files belonging to this
+   * provided file group.
+   */
+  public static Option<Pair<String, List<String>>> getFileSliceForFileGroupFromDeltaCommit(
```

Comment:
How would this be compatible if we evolve HoodieCommitMetadata.avsc? We should avoid doing this json tree walking and read the content with defined avro schemas.

Reply:
you're right  that maybe need to adapt manually when evolve HoodieCommitMetadata.
I have tried to use some simple changed to achieve this, but i failed. another reply about this: https://github.com/apache/hudi/pull/5885#discussion_r925112104
so can we use another ticket to trace this?

Reply:
i'm ok with moving forward with a proper refactoring later, if you can annotate this method with a jira and take it up soon after first version.

Reply:
https://issues.apache.org/jira/browse/HUDI-4822

Reply:
Can you a leave a TODO in code for the refactor and link jira?

Reply:
OK